src/core/Akka.Remote/MessageSerializer.cs

```diff
@@ -41,16 +42,16 @@ public static object Deserialize(ActorSystem system, SerializedMessage messagePr
         /// <returns>SerializedMessage.</returns>
         public static SerializedMessage Serialize(ActorSystem system, Address address, object message)
         {
-            Serializer serializer = system.Serialization.FindSerializerFor(message);
+            var objectType = message.GetType();
+            Serializer serializer = system.Serialization.FindSerializerForType(objectType);
 
             var serializedMsg = new SerializedMessage
             {
                 Message = ByteString.CopyFrom(serializer.ToBinaryWithAddress(address, message)),
```

Comment:
If I change it to `ToBinary` I could save another 128 bytes of allocations per call

|                                  Method |      Mean |     Error |    StdDev |  Gen 0 | Allocated |
|---------------------------------------- |----------:|----------:|----------:|-------:|----------:|
|   MessageSerializer_Serialize_Primitive | 46.153 us | 0.2032 us | 0.1901 us | 0.1343 |    6161 B |
|     MessageSerializer_Serialize_Complex | 61.302 us | 0.3956 us | 0.3089 us | 0.2441 |   10473 B |